packages/pynumaflow/pynumaflow/sinker/servicer/utils.py

```diff
@@ -32,12 +34,36 @@ def build_sink_response(rspn: Response) -> sink_pb2.SinkResponse.Result:
         return sink_pb2.SinkResponse.Result(id=rid, status=sink_pb2.Status.SUCCESS)
     elif rspn.fallback:
         return sink_pb2.SinkResponse.Result(id=rid, status=sink_pb2.Status.FALLBACK)
+    elif rspn.on_success:
+        return sink_pb2.SinkResponse.Result(
+            id=rid,
+            status=sink_pb2.Status.ON_SUCCESS,
+            on_success_msg=build_on_success_message(rspn.on_success_msg),
+        )
     else:
         return sink_pb2.SinkResponse.Result(
             id=rid, status=sink_pb2.Status.FAILURE, err_msg=rspn.err
         )
 
 
+def build_on_success_message(
+    msg: Optional[Message],
+) -> Optional[sink_pb2.SinkResponse.Result.Message]:
+    if not msg:
+        return None
```

Comment:
This works, but its for all kind of truthiness like 0, `False`, empty string etc.
`if msg is None` is the correct approach.